distribution/tools/entitlement-agent/src/main/java/org/elasticsearch/entitlement/instrumentation/MethodKey.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package org.elasticsearch.entitlement.instrumentation;
+
+import org.objectweb.asm.Type;
+
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+
+/**
+ *
+ * @param className the "internal name" of the class: includes the package info, but with periods replaced by slashes
+ * @param methodName
+ * @param voidDescriptor the method descriptor, but with the return type replaced by {@code V}
+ *                       because the return type isn't consulted in the lookup process anyway
+ * @param isStatic
+ */
+public record MethodKey(String className, String methodName, String voidDescriptor, boolean isStatic) {
+    /**
+     * @return a {@link MethodKey} suitable for looking up the given {@code targetMethod} in the entitlements trampoline
+     */
+    public static MethodKey forTargetMethod(Method targetMethod) {
+        Type actualType = Type.getMethodType(Type.getMethodDescriptor(targetMethod));
+        String voidDescriptor = Type.getMethodDescriptor(Type.VOID_TYPE, actualType.getArgumentTypes());
```

Comment:
Rather than storing the descriptor with `void` at the start (which is wrong), could you store it without a return type at all, then add in the `void` when doing the actual lookup?

Reply:
Trying to remember why I did it this way. It was a couple of weeks ago. 🤔

Reply:
Oh ok I know what it is. We need to do both, so they match!

At configuration time, we do not want to express an opinion on the return type. At instrumentation time, we don't want the return type to be considered. Hence, we stamp over both of them with void so they match, making the `HashMap` lookup work.